samples/matter/common/src/app/migration_manager.cpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "migration_manager.h"
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+namespace Nrf::Matter
+{
+namespace Migration
+{
+#ifdef CONFIG_CHIP_CRYPTO_PSA
+	CHIP_ERROR MoveOperationalKeysFromKvsToIts(chip::PersistentStorageDelegate *storage,
+						   chip::Crypto::OperationalKeystore *keystore)
+	{
+		CHIP_ERROR err = CHIP_NO_ERROR;
+
+		VerifyOrReturnValue(keystore && storage, CHIP_ERROR_INVALID_ARGUMENT);
+
+		/* Initialize the obsolete Operational Keystore*/
+		chip::PersistentStorageOperationalKeystore obsoleteKeystore;
+		err = obsoleteKeystore.Init(storage);
+		VerifyOrReturnError(err == CHIP_NO_ERROR, err);
+
+		/* Migrate all obsolete Operational Keys to PSA ITS */
+		for (const chip::FabricInfo &fabric : chip::Server::GetInstance().GetFabricTable()) {
+			err = keystore->MigrateOpKeypairForFabric(fabric.GetFabricIndex(), obsoleteKeystore);
+			if(CHIP_NO_ERROR != err) {
+#ifdef CONFIG_NCS_SAMPLE_MATTER_FACTORY_RESET_ON_KEY_MIGRATION_FAILURE
+				chip::Server::GetInstance().ScheduleFactoryReset();
```

Comment:
Shouldn't we return after scheduling factory reset? If we have more than one fabric, theoretically we may end scheduling factory reset several times, which is pointless, right?